packages/core/types/src/http/product/common.ts

```diff
@@ -0,0 +1,192 @@
+import { BaseFilterable, OperatorMap } from "../../dal"
+import { BaseProductCollection } from "../collection/common"
+
+export type ProductStatus = "draft" | "proposed" | "published" | "rejected"
+
+export interface BaseProduct {
+  id?: string
+  title?: string
+  handle?: string
+  subtitle?: string
+  description?: string
+  is_giftcard?: boolean
+  status?: ProductStatus
+  thumbnail?: string
+  width?: number
+  weight?: number
+  length?: number
+  height?: number
+  origin_country?: string
+  hs_code?: string
+  mid_code?: string
+  material?: string
+  collection?: BaseProductCollection
+  collection_id?: string
+  categories?: BaseProductCategory[]
+  type?: BaseProductType
+  type_id?: string
+  tags?: BaseProductTag[]
+  variants?: BaseProductVariant[]
+  options?: BaseProductOption[]
+  images?: BaseProductImage[]
+  discountable?: boolean
+  external_id?: string
+  created_at?: string
+  updated_at?: string
+  deleted_at?: string
+  metadata?: Record<string, unknown>
```

Comment:
todo: re work the nullability everywhere

Reply:
@adrien2p I don't think we need to have explicit `null` values as typings in the `http` response, what is the benefit of it? These models won't be used for write operations, we'll have separate typings for that

Reply:
I think they are important in the sense that they means different thing, undefined probably mean that the property is not there, null means it has been returned but does not have any values, so you can set some props to null and when you get them back they might be present but null, and it is not because they are not undefined that they are not null if that make sense

Reply:
Oops missed addressing this comment. I think it makes sense, just one issue currently is that pretty much anything can be null, do we just set it on all fields? Not sure how we approach this..

Reply:
Not sure everything can be null when it is expected to be present. But props like metadata for example are nullable while they are present as well as can be optional. Same for deleted at for example. I think we can start with the obvious ones like the two mentioned here and then we can refine along the water if the other ones are not obvious wdyt ?

Reply:
Yeah sure, let's go with that

Reply:
Added some already, we can refine as we go 👍

Reply:
If we can add that to the limit too, if limit is null explicitly then we return everything otherwise we apply the value and if not provided we apply the default limit wdyt?